tests/integration/test_tls.py

```diff
@@ -0,0 +1,733 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from time import sleep
+
+import pytest
+from juju.application import Application
+from pytest_operator.plugin import OpsTest
+
+from literals import INTERNAL_USER, PEER_RELATION, TLSType
+
+from .helpers import (
+    APP_NAME,
+    get_certificate_from_unit,
+    get_cluster_endpoints,
+    get_cluster_members,
+    get_juju_leader_unit_name,
+    get_key,
+    get_secret_by_label,
+    put_key,
+)
+
+logger = logging.getLogger(__name__)
+
+TLS_NAME = "self-signed-certificates"
+NUM_UNITS = 3
+TEST_KEY = "test_key"
+TEST_VALUE = "42"
+CERTIFICATE_EXPIRY_TIME = 90
+
+
+@pytest.mark.runner(["self-hosted", "linux", "X64", "jammy", "large"])
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy_with_tls(ops_test: OpsTest) -> None:
+    """Build the charm-under-test and deploy it with three units.
+
+    The initial cluster should be formed and accessible.
+    """
+    assert ops_test.model is not None, "Model is not set"
+    # Deploy the TLS charm
+    tls_config = {"ca-common-name": "etcd"}
+    await ops_test.model.deploy(TLS_NAME, channel="edge", config=tls_config)
+    # Build and deploy charm from local source folder
+    etcd_charm = await ops_test.build_charm(".")
+    model = ops_test.model_full_name
+    assert model is not None, "Model is not set"
+    # Deploy the charm and wait for active/idle status
+    logger.info("Deploying the charm")
+    await ops_test.model.deploy(etcd_charm, num_units=NUM_UNITS)
+
+    # enable TLS and check if the cluster is still accessible
+    logger.info("Integrating peer-certificates and client-certificates relations")
+    await ops_test.model.integrate(f"{APP_NAME}:peer-certificates", TLS_NAME)
+    await ops_test.model.integrate(f"{APP_NAME}:client-certificates", TLS_NAME)
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000)
+
+
+@pytest.mark.runner(["self-hosted", "linux", "X64", "jammy", "large"])
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_tls_enabled(ops_test: OpsTest) -> None:
+    """Check if the TLS has been enabled on app startup."""
+    # check if all units have been added to the cluster
+    assert ops_test.model is not None, "Model is not set"
+    model = ops_test.model_full_name
+    assert model is not None, "Model is not set"
```

Comment:
You're already asserting on it in the previous test `test_build_and_deploy_with_tls`.
we can remove this here and in all subsequent tests